Source/WebCore/editing/EditingStyle.cpp

```diff
@@ -1430,6 +1431,21 @@ void EditingStyle::removeStyleFromRulesAndContext(StyledElement& element, Node*
         if (!computedStyle->m_mutableStyle->getPropertyCSSValue(CSSPropertyBackgroundColor))
             computedStyle->m_mutableStyle->setProperty(CSSPropertyBackgroundColor, CSSValueTransparent);
 
+        // If white-space has changed, do not remove white-space longhand values.
+        // They are necessary for reconstructing the corresponding white-space shorthand value.
+        auto whiteSpaceCollapse = m_mutableStyle->getPropertyCSSValue(CSSPropertyWhiteSpaceCollapse);
+        auto whiteSpaceCollapseContext = computedStyle->m_mutableStyle->getPropertyCSSValue(CSSPropertyWhiteSpaceCollapse);
```

Comment:
previousWhiteSpaceCollapse?

Reply:
I feel like "previous" might not be the most suitable term here. Basically, `whiteSpaceCollapseContext` is the style of the parent element while `whiteSpaceCollapse` is the style of the child element. Maybe something like `localWhiteSpaceCollapse`?